### PR TITLE
IGNITE-18570 Use network serialization for Meta Storage API

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-18570

### CODE REVIEW

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/command/MetastorageCommandsMessageGroup.java [hunk 1]
```diff
@@ -21,41 +21,13 @@
...
-    short COMPOUND_CONDITION_INFO = 5;
-
-    /** Message type for {@link StatementResultInfo}. */
-    short STATEMENT_RESULT_INFO = 6;
-
-    /** Message type for {@link IfInfo}. */
-    short IF_INFO = 7;
-
     /** Message type for {@link InvokeCommand}. */
     short INVOKE = 10;
```
Comment (SammyVimes) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/command/MetastorageCommandsMessageGroup.java:32]: Why not 0?

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/MetaStorageMessageGroup.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.apache.ignite.internal.metastorage.dsl.SimpleCondition.RevisionCondition;
+import org.apache.ignite.internal.metastorage.dsl.SimpleCondition.ValueCondition;
+import org.apache.ignite.internal.metastorage.dsl.Statement.IfStatement;
+import org.apache.ignite.internal.metastorage.dsl.Statement.UpdateStatement;
+import org.apache.ignite.network.annotations.MessageGroup;
+
+/**
+ * Message group for Meta Storage messages.
+ */
+@MessageGroup(groupType = 222, groupName = "MetaStorageMessages")
```
Comment (SammyVimes) [inline modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/MetaStorageMessageGroup.java:29]: Why 222?

File: modules/network/src/main/java/org/apache/ignite/internal/network/netty/OutboundEncoder.java [hunk 3]
```diff
@@ -96,14 +96,12 @@ private NetworkMessageChunkedInput(
...
+            List<ClassDescriptorMessage> outDescriptors = outObject.descriptors().stream()
+                    .filter(classDescriptorMessage -> !serializationService.isDescriptorSent(classDescriptorMessage.descriptorId()))
+                    .collect(Collectors.toList());
 
-                descriptors = descriptors.stream()
-                        .filter(classDescriptorMessage -> !serializationService.isDescriptorSent(classDescriptorMessage.descriptorId()))
-                        .collect(Collectors.toList());
-
-                this.descriptors = MSG_FACTORY.classDescriptorListMessage().messages(descriptors).build();
+            if (!outDescriptors.isEmpty()) {
```
Comment (SammyVimes) [inline modules/network/src/main/java/org/apache/ignite/internal/network/netty/OutboundEncoder.java:103]: Nice Catch!

**Review by SammyVimes — APPROVED** [review]

File: modules/network/src/main/java/org/apache/ignite/internal/network/netty/OutboundEncoder.java (hunk 3 above)
Comment (sashapolo) [reply modules/network/src/main/java/org/apache/ignite/internal/network/netty/OutboundEncoder.java:103]: Noticed this behavior during a debugging session

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/MetaStorageMessageGroup.java (hunk 2 above)
Comment (sashapolo) [reply modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/MetaStorageMessageGroup.java:29]: No particular reason

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/command/MetastorageCommandsMessageGroup.java (hunk 1 above)
Comment (sashapolo) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/command/MetastorageCommandsMessageGroup.java:32]: To reduce the amount of changes

**Pull request merged**